packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart

```diff
@@ -174,10 +181,21 @@ class MethodChannelImagePicker extends ImagePickerPlatform {
 
     final String? path = result['path'];
 
+    // In this case, multiRetrieve is invoked.
+    if (!isRetrieve) {
+      final pathList = result['pathList'];
+      if (pathList != null) {
+        for (String path in pathList) {
+          pickedFileList.add(PickedFile(path));
+        }
+      }
+    }
+
     return LostData(
       file: path != null ? PickedFile(path) : null,
       exception: exception,
       type: retrieveType,
+      fileList: pickedFileList.isNotEmpty ? pickedFileList : null,
```

Comment:
If you define `pickedFileList` as `List<PickedFile>? pickedFileList = null;` (especially because you already made it nullable) and then in `if (pathList != null) {` add `pickedFileList = [];` you don't need this check anymore.
That also means that you will return an empty list when it actually is an empty list instead of returning null.